pkg/workflows/management/core_components.go

```diff
@@ -92,10 +92,13 @@ func runUpgradeCoreComponents(ctx context.Context, commandContext *task.CommandC
 		commandContext.SetError(err)
 		return err
 	}
-	commandContext.ClusterSpec.Cluster.SetManagementComponentsVersion(commandContext.ClusterSpec.EKSARelease.Spec.Version)
+
+	// ManagedFields must be nil to perform the server-side apply or a BadRequest error is thrown.
+	commandContext.CurrentClusterSpec.Cluster.ManagedFields = nil
```

Comment:
Unsure how i feel about this one, as it could be a problem if `ManagedFields` is used later on in the workflow for something. However, since it's typically used for kubernetes internal house keeping and server side apply may update this anyway (meaning it would have to be read again), I think this does not pose an issue.